tree/ntuple/v7/test/ntuple_zip.cxx

```diff
@@ -84,3 +84,118 @@ TEST(RNTupleZip, Large)
+         std::make_unique<RPageSinkFile>("ntuple1", *file, overrideCompression));
+      ntuple1->Fill();
+   }
+   // test using RNTupleWriter::Append (which calls the RPageSinkFile TFile& constructor)
+   {
+      auto make_ntuple = [&](std::string name, const RNTupleWriteOptions& opt) {
+         auto model = RNTupleModel::Create();
+         auto field = model->MakeField<float>("field");
+         return RNTupleWriter::Append(std::move(model), name, *file, opt);
+      };
+      // ntuple2 has the default compression setting and so will use the file's setting
+      auto ntuple2 = make_ntuple("ntuple2", RNTupleWriteOptions());
+      // ntuple3 has a specific compression setting and will use it
+      auto ntuple3 = make_ntuple("ntuple3", overrideCompression);
+      // ntuple4 has the default explicity set to avoid using the file's setting
+      RNTupleWriteOptions defaultCompression;
+      defaultCompression.SetCompression(404);
+      auto ntuple4 = make_ntuple("ntuple4", defaultCompression);
+      ntuple2->Fill();
+      ntuple3->Fill();
+      ntuple4->Fill();
+   }
+   file.reset();
+
+   std::ostringstream oss;
+   // ... ntuple1 uses a specific compression level
+   auto ntuple1 = RNTupleReader::Open("ntuple1", fileGuard.GetPath());
+   ntuple1->PrintInfo(ROOT::Experimental::ENTupleInfo::kStorageDetails, oss);
+   EXPECT_THAT(oss.str(), testing::HasSubstr("Compression: 505"));
+   oss.str("");
+   // ... ntuple2 uses the TFile's compression level
+   auto ntuple2 = RNTupleReader::Open("ntuple2", fileGuard.GetPath());
+   ntuple2->PrintInfo(ROOT::Experimental::ENTupleInfo::kStorageDetails, oss);
+   EXPECT_THAT(oss.str(), testing::HasSubstr("Compression: 101"));
+   oss.str("");
+   // ... ntuple3 uses a specific compression level
+   auto ntuple3 = RNTupleReader::Open("ntuple3", fileGuard.GetPath());
+   ntuple3->PrintInfo(ROOT::Experimental::ENTupleInfo::kStorageDetails, oss);
+   EXPECT_THAT(oss.str(), testing::HasSubstr("Compression: 505"));
+   oss.str("");
+   // ... ntuple4 uses a specific compression level
+   auto ntuple4 = RNTupleReader::Open("ntuple4", fileGuard.GetPath());
+   ntuple4->PrintInfo(ROOT::Experimental::ENTupleInfo::kStorageDetails, oss);
+   EXPECT_THAT(oss.str(), testing::HasSubstr("Compression: 404"));
+   oss.str("");
+}
+
+TEST(RNTupleZip, TFileCompressionNotUpdated)
+{
+   FileRaii fileGuard("test_ntuple_zip_tfile_comp_not_updated.root");
+   auto file = std::make_unique<TFile>(fileGuard.GetPath().c_str(), "RECREATE", "", 101);
+   {
+      auto model = RNTupleModel::Create();
+      auto field = model->MakeField<float>("field");
+      // ntuple is created when TFile has compression setting 101
+      auto ntuple = std::make_unique<RNTupleWriter>(std::move(model),
+         std::make_unique<RPageSinkFile>("ntuple", *file, RNTupleWriteOptions()));
+      // if the TFile compression is later adjusted, this won't be picked up by the ntuple
+      file->SetCompressionSettings(404);
```

Comment:
I am now not sure if this PR's current approach is viable, because if users adjust the TFile settings after the RNTuple is constructed the change will not be applied to the RNTuple.

Reply:
In TTree, we use `ROOT::RCompressionSetting::ELevel::kInherit` to indicate that (*at compression time*) the compression level should be grabbed from the TFile.

Reply:
Good point. I think RNTuple should do the same.

Reply:
I think this is resolved by acd1a9f